docs/docs/usage/dependency.md

```diff
@@ -36,23 +35,44 @@ For details of the meaning of each field in `pyproject.toml`, please refer to [P
 
 ```console
 $ pdm add requests
-$ pdm add -d pytest
 ```
 
 `pdm add` can be followed by one or several dependencies, and the dependency specification is described in
 [PEP 508](https://www.python.org/dev/peps/pep-0508/).
-There are two groups of dependencies: packages will be added to `project.dependencies` by default or `project.dev-dependencies`
-if `-d/--dev` option is passed to the `pdm add` command.
 
-PDM also allows extra dependency groups by providing `-s/--section <name>` option, and the dependencies will appear in
-`project.optional-dependencies.<name>` in the project file, respectively.
+PDM also allows extra dependency groups by providing `-s/--section <name>` option, and those dependencies will go to
+`[project.optional-dependencies.<name>]` table in the project file, respectively.
 
 After that, dependencies and sub-dependencies will be resolved properly and installed for you, you can view `pdm.lock` to see
 the resolved result of all dependencies.
 
 Local packages can be installed in [editable mode](https://pip.pypa.io/en/stable/reference/pip_install/#editable-installs)
 (just like `pip install -e <local project path>` would) using `pdm add -e/--editable <local project path>`.
 
+## Add development only dependencies
+
+_New in 1.5.0_
+
+PDM also supports defining groups of dependencies that are useful for development. They can be classified as different groups,
+e.g. some for testing and others for linting. We usually don't want these dependencies appear in the distribution's metadata
+so using `optional-dependencies` is probably not a good idea. We can define them as development dependencies:
+
+```console
+$ pdm add -ds test pytest
+```
+
+This will result in a pyproject.toml as following:
+
+```toml
+[tool.pdm.dev-dependencies]
+test = ["pytest"]
+```
+
+For backward-compatibility, if `-s/--section` is not given, dependencies will go to `dev` group under `[tool.pdm.dev-dependencies]` by default.
+
+!!! NOTE
+    The group names in `[tool.pdm.dev-dependencies]` MUST NOT conflict with those in `[project.optional-dependencies]`.
```

Comment:
What exactly is considered a "conflict"?
* A group name appears in both sections (independent of its content)?
* A group name appears in both sections with different content?

Reply:
A group name appears in both sections (independent of its content)

This is required because currently the `sections` in `pdm.lock` can't distinguish whether from dev deps or optional deps

Reply:
Then I would explicitly state it like this:

> A group name MUST NOT appear in both `[tool.pdm.dev-dependencies]` and `[project.optional-dependencies]`.

... or something similar.